lib/route/nh.c

```diff
@@ -258,6 +260,28 @@ int rtnl_nh_get_family(struct rtnl_nh *nexthop)
 	return nexthop->nh_family;
 }
 
+int rtnl_nh_set_group_type(struct rtnl_nh *nexthop, uint16_t group_type)
+{
+	if (!nexthop)
+		return -NLE_INVAL;
+
+	nexthop->nh_group_type = group_type;
+	nexthop->ce_mask |= NH_ATTR_GROUP_TYPE;
+
+	return 0;
+}
+
+int rtnl_nh_get_group_type(struct rtnl_nh *nexthop)
+{
+	if (!nexthop)
+		return -NLE_INVAL;
+
+	if (!(nexthop->ce_mask & NH_ATTR_GROUP_TYPE))
+		return -NLE_INVAL;
+
+	return (int)nexthop->nh_group_type;
```

Comment:
in this case, we encode the error value and the success value in `int`. I think that is fine for `NHA_GROUP_TYPE`.

At least, as long as we are sure that this value is always just an `u16` (that fits inside the positive range of `int`).

In that case, `nh_group_type` in `struct rtnl_nh` is wrongly defined as `uint32_t`. Could you please fix that too?

Or do you think there might be cases where it's not `u16`? In that case, the getter would have to return the value and the error code separately.